File: equilibratore/equilibratore.py
#definisco il metodo
def equilibratore(x, y):
    sum = x + y
    x = sum / 2
    y = sum - x
    #python mi permette di ritornare più di un valore
    return x, y

x = int(input("Inserisci il primo numero: "))
y = int(input("Inserisci il secondo numero: "))

#chiamo il metodo
x , y = equilibratore(x, y)

print("I numeri sono stati equilibrati a", x, "e", y)

File: equilibratore/equilibratore.java
import java.util.Scanner;

public class equilibratore{
    
    public static void main(String[] args) {
    
        //per prendere in ingresso i due numeri
        Scanner scanner = new Scanner(System.in);

        System.out.println("Inserisci il primo numero: ");
        Integer n1 = scanner.nextInt();

        System.out.println("Inserisci il secondo numero: ");
        Integer n2 = scanner.nextInt();

        Integer sum = n1 + n2;

        n1 = sum / 2;
        n2 = sum - n1;

        System.out.println("I due numeri sono stati equilibrati a " + n1 + " e " + n2);

    }

}
